homeassistant/helpers/entity_platform.py

```diff
@@ -197,14 +199,15 @@ async def _async_setup_platform(
                 await asyncio.shield(task)
 
             # Block till all entities are done
-            if self._tasks:
+            while self._tasks:
```

Comment:
Feels a bit unrelated. I do understand it and I think it is correct, but can't this cause issues? Say a integration that starts task for use after setup during setup.

I'm just thinking we might get regressions with this. But I suppuse anything found should be fixed.

Reply:
Will keep an eye out for regressions 👍